src/Integration.UnitTests/State/StateManagerTests.cs

```diff
@@ -0,0 +1,411 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="StateManagerTests.cs" company="SonarSource SA and Microsoft Corporation">
+//   Copyright (c) SonarSource SA and Microsoft Corporation.  All rights reserved.
+//   Licensed under the MIT License. See License.txt in the project root for license information.
+// </copyright>
+//-----------------------------------------------------------------------
+
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using SonarLint.VisualStudio.Integration.Resources;
+using SonarLint.VisualStudio.Integration.Service;
+using SonarLint.VisualStudio.Integration.State;
+using SonarLint.VisualStudio.Integration.TeamExplorer;
+using SonarLint.VisualStudio.Integration.WPF;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Windows.Input;
+
+namespace SonarLint.VisualStudio.Integration.UnitTests.State
+{
+    [TestClass]
+    public class StateManagerTests
+    {
+        [TestInitialize]
+        public void TestInitialize()
+        {
+            ThreadHelper.SetCurrentThreadAsUIThread();
+        }
+
+        #region Tests
+        [TestMethod]
+        public void StateManager_ArgsCheck()
+        {
+            Exceptions.Expect<ArgumentNullException>(() => new StateManager(null, new TransferableVisualState()));
+            Exceptions.Expect<ArgumentNullException>(() => new StateManager(new ConfigurableHost(), null));
+        }
+
+        [TestMethod]
+        public void StateManager_SetProjectsUIThread()
+        {
+            // Setup
+            var section = ConfigurableConnectSection.CreateDefault();
+            ConfigurableUserNotification notifications = (ConfigurableUserNotification)section.UserNotifications;
+            ConfigurableHost host = new ConfigurableHost();
+            StateManager testSubject = this.CreateTestSubject(host);
+            host.VisualStateManager = testSubject;
+            section.ViewModel.State = testSubject.ManagedState;
```

Comment:
We do this (`section.ViewModel.State = testSubject.ManagedState`) in every test setup, move it in to the create helper.